### ISSUE TITLE
Expose interface speed

### ISSUE THREAD

**Issue by kaihendry** [nshttpd/mikrotik-exporter#44]
I would like to query negotiated speed. For some reason at home I have dodgy long cables, so would be nice to see issues via the exporter.

    /interface ethernet print detail where speed!=1Gbps

**Comment by nshttpd** [comment]
Do you want the default speed without auto-negotiation (which is what the print detail shows in that `speed` value) or the actual line speed from the `/interface ethernet monitor` output `rate` value? Let me play and see if the monitor one can be queried via an API call and what it returns.

**Comment by kaihendry** [comment]
Yes, sorry the **monitor** aspect, the one that shows the real output rate instead of the defaults as I think `/interface ethernet print detail` shows.

**Comment by nshttpd** [comment]
@kaihendry can you check out 1.0.10-DEVEL version (binaries on github and container on docker hub, brought to you by CircleCI) and let me know if that works for you. I created a monitor collector and it's either `monitor: true` in a config file or `-with-monitor` on the CLI to enable it.

It'll report back if the ethernet interface is up via : `mikrotik_monitor_status` 0 down 1 up

`mikrotik_monitor_rate` for the interface speed 10, 100, 1000  .. which I actually found some of my stuff running at 100 Mbps instead of 1Gbps that I need to dig into tomorrow and figure out what's up with that.

`mikrotik_monitor_full_duplex` .. 0 half 1 full

**Comment by kaihendry** [comment]
Wonder how to work out what version I'm running after updating?

https://s.natalian.org/2019-08-12/version.txt

**Comment by nshttpd** [comment]
There are pre-build binaries :

https://github.com/nshttpd/mikrotik-exporter/releases/tag/1.0.10-DEVEL

or if you want to build this specific one you'll have to pull it from github and build the `nshttpd/monitor-api` branch.

and I'll put in a issue to add a version flag.

**Comment by kaihendry** [comment]
`curl -LO https://github.com/nshttpd/mikrotik-exporter/releases/download/1.0.10-DEVEL/mikrotik-exporter_linux_amd64` bizarrely doesn't work on my [muslc based](https://wiki.voidlinux.org/Musl) nuc!

https://s.natalian.org/2019-08-14/go-binaries.mp4

   -bash: ./mikrotik-exporter_linux_amd64: No such file or directory

**Comment by kaihendry** [comment]
How do you create the pre-built binaries btw? **CGO_ENABLED=0** needs to be on I think: https://www.reddit.com/r/voidlinux/comments/cq6lkh/can_i_expect_amd64_go_static_binaries_not_to_be/ewuegh4/

Do you not use https://github.com/goreleaser/goreleaser ?

**Comment by nshttpd** [comment]
I'm using gox https://github.com/mitchellh/gox .. Check out the Makefile for the deploy step. 

I just rebuilt all of them with the `CGO_ENABLED=0` environment var set.

**Comment by kaihendry** [comment]
Thanks, works ! 

```
hendry@knuckles /tmp $ curl -LO https://github.com/nshttpd/mikrotik-exporter/releases/download/1.0.10-DEVEL/mikrotik-exporter_linux_amd64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   620    0   620    0     0   1479      0 --:--:-- --:--:-- --:--:--  1479
100 11.0M  100 11.0M    0     0  2753k      0  0:00:04  0:00:04 --:--:-- 3571k
hendry@knuckles /tmp $ ./mikrotik-exporter_linux_amd64
{"level":"error","msg":"Could not load config: missing required param for single device configuration","time":"2019-08-19T12:29:16+08:00"}
```


Sidenote: Wonder how gox compares to goreleaser

**Comment by kaihendry** [comment]
<img src="https://s.natalian.org/2019-08-19/1566203997_2560x1440.png">

**Comment by nshttpd** [comment]
cool. i'll merge to master and close this after ${WORK} when I get home.

**Comment by nshttpd** [comment]
merged and actual 1.0.10 release cut with this feature in there.
